hadoop-tools/hadoop-aliyun/src/main/java/org/apache/hadoop/fs/aliyun/oss/AliyunOSSFileSystemStore.java

```diff
@@ -527,17 +530,14 @@ protected OSSListRequest createListObjectsRequest(String prefix,
    * @return This method returns null if the key is not found.
    */
   public InputStream retrieve(String key, long byteStart, long byteEnd) {
-    try {
-      GetObjectRequest request = new GetObjectRequest(bucketName, key);
-      request.setRange(byteStart, byteEnd);
-      InputStream in = ossClient.getObject(request).getObjectContent();
-      statistics.incrementReadOps(1);
-      return in;
-    } catch (OSSException | ClientException e) {
-      LOG.error("Exception thrown when store retrieves key: "
-              + key + ", exception: " + e);
-      return null;
+    GetObjectRequest request = new GetObjectRequest(bucketName, key);
+    if (useStandardHttpRangeBehavior) {
+      request.addHeader("x-oss-range-behavior", "standard");
```

Comment:
1. can you document this
2. are multiple ranges supported? aws s3 doesnt, though dell storage does

Reply:
Ok, I will add some comments. OSS does not support multiple ranges.

Reply:
oh well, unfortunate but not unusual